### ISSUE TITLE
Sometimes HTMLVideoElement not appears

### ISSUE THREAD

**Issue by Bruno92400** [twilio/twilio-video.js#427]
Hi,

Sometimes all work fine.
And sometimes, after i create room and attach tracks of an other participant his readyState of  HTMLVideoElement (video track) stay to 0 and the video not appears.

What happend in this case?

Is there a possibility to reload the HTMLVideoElement or the video track please?

Best regards,

**Comment by markandrus** [comment]
Hi @Bruno92400,

Are you using the AudioTrack and VideoTrack `attach` methods to attach to HTMLVideoElements? It's possible the video will not appear if

* The VideoTrack is disabled on the remote side (i.e., the publisher disabled the VideoTrack).
* The connection to the Room is bad (hence, no video frames are being received)—listen to Room's "reconnecting" and "reconnected" events.

If you have a Room SID, we could help look into it.

Best,
Mark

**Comment by Bruno92400** [comment]
Yes, i use `attach` and `detach` methods.

I can recreate the ploblem with 2 PC of mine after many test to enter in the room so it is not the publisher.

Ok, i try to listen to Room's "reconnecting" and "reconnected" events, thank you.

This is one of the Room SID with problem : RMe19250794ac0ae45f375148648ae65c4

Best,
Bruno

**Comment by markandrus** [comment]
Thanks, @Bruno92400.

Actually, the Room has a lot of history on it. It may be better to share particular Participant SIDs who experienced the problem. For example, in that Room, I see that PA53cac10930e476c56cf8b277a8caaf27 and PA9f88d56bf878f4d4f89263c4c2b96bd2 experienced ICE failures, indicating that their connection to the Room went bad. If either of them were publishing the VideoTracks that failed to playback, that would explain it.

Mark

**Comment by Bruno92400** [comment]
It happends now in Room SID :  RM263f6e7454a98aa9b816ba585982ccd3
The user : PAd8c1839448a1a8212529bed7cfe7cfda cannot see the video of PA7324d3996a2c45800535965bae4d23f2

**Comment by Bruno92400** [comment]
After test with the problem `room` `reconnecting` and `reconnected` say nothing.

Is there a way to stop and start (so restart) the track that have the problem? Maybe a solution like that works?

I try a lot of things, this is my last try : 
      ```
$scope.reloadVideos = function () {
        // HTMLVideoElement
        $('video').each(function() {
          console.log(this.readyState);
          if (this.readyState < 4) {
            console.log(this.constructor.name);
            console.log(this);

            var videoParticipantId = $(this).parent().attr('id');

            $($scope.videoTracks[videoParticipantId].detach()).remove();
            $(this)
              .parent()
              .append($($scope.videoTracks[videoParticipantId].attach()))
            ;

            // $scope.videoConf.room.localParticipant.unpublishTracks([$scope.videoTracks[videoParticipantId]]);
            // $scope.videoConf.room.localParticipant.publishTrack($scope.videoTracks[videoParticipantId]);
          }
        });
      }
```

**Comment by Bruno92400** [comment]
And now, Firefox to Firefox or safari it never works. The same problem but always.
It works only Chrome to Chrome or Chrome to Firefox.

What can i do, our customers are very dissatisfied.

**Comment by ZelStudio** [comment]
Today, iOS 11 and Safari or Chorme on Mac OS have the same issue, no video or it's working and it cut after few minutes.
